### DISCUSSION TITLE
Xdotool

### DISCUSSION THREAD

**Discussion by fahad5359** [microsoft/WSL#9704] [General]
I was wonderig if wsl supports the use of **xdotool**?

The xdotool does not work for me. 

 I installed the xming server, and its tools, but when i try to do a command regaring the xdotool and press ENTER, it does not do any thing just a new line. No errors nothing...

Thank you.

**Comment by elsaco** [comment]
`xdotool` works fine in WSL. Please specify what WSL version and OS are you using. While `xming` might work an X server is not necessary with the latest WSL versions. The service is provided by WSLg on Win10 & Win11.

Sample output showing some window geometry:

```
elsaco@RIPPER:~$ xdotool getwindowgeometry 6291466
Window 6291466
  Position: 1835,325 (screen: 0)
  Geometry: 150x100
```

**Reply by fahad5359** [reply]
Thank you i manged to display  some results using 
`sudo apt install xterm`

**Comment by cyborgdennett** [comment]
I'm trying `xdotool key ctrl+plus` but nothing happens

**Comment by ubaldot** [comment]
I just tried: `xdotool search --onlyvisible --name yt7 windowactivate` and I expected the window to be brought in front, but nothing happened. However, `xdotool search --onlyvisible --name yt7 windowsize 800 1000` works well. Working on Win10 with Ubuntu 22.04 installed.

**Comment by ubaldot** [comment]
There are actually a number of command that don't seem to work on WSL. If I use `xdotool search --onlyvisible --name yt7` I get `8388611`, but then I got the following: 

```
$ xdotool windowfocus 8388611
```
-> Nothing happens

```
$ xdotool windowactivate 8388611
```
-> Nothing happens

```
$ xdotool getwindowfocus
xdo_get_focused_window_sane failed (code=1)
xdo_focus_window reported an error
```
